precode.sh

```diff
@@ -7,37 +7,70 @@
 #   dir3
 #       dir4
 
+mkdir task && mkdir task/dir{1..3} && mkdir -p task/dir3/dir4
+
 # изменяем текущую директорию на task
 
+cd task
+
 # создаём пустой файл task/dir2/empty
 
+touch dir2/empty
+
 # создаём файл task/dir2/hello.sh с таким содержанием:
 # #!/bin/bash
 # echo "$1, привет!"
 
+echo '#!/bin/bash'> dir2/hello.sh && echo 'echo "$1, привет!" '>> dir2/hello.sh
+
 # устанавливаем для task/dir2/hello.sh права rwxrw-r--
 
+chmod 764 dir2/hello.sh
+
 # сохраняем список файлов task/dir2 в task/dir2/list.txt
 
+ls dir2 > dir2/list.txt
+
 # копируем содержимое каталога task/dir2 в каталог task/dir3/dir4
 
+cp -R dir2/. dir3/dir4/
```

Comment:
Это не мешает, но можно убрать
```suggestion
cp -R dir2 dir3/dir4
```